tests/test_response.py

```diff
@@ -235,13 +235,29 @@ def test_chunked_streaming_returns_correct_content(streaming_app):
     assert response.text == "foo,bar"
 
 
+@pytest.mark.asyncio
+async def test_chunked_streaming_returns_correct_content_asgi(streaming_app):
+    request, response = await streaming_app.asgi_client.get("/")
+    assert response.text == "4\r\nfoo,\r\n3\r\nbar\r\n0\r\n\r\n"
```

Comment:
So turns out this is wrong.
The response.text should be "foo,bar" just like in "test_chunked_streaming_returns_correct_content" non-asgi test.
The bug #1964 is causing the extra lengths and `\r\n` to be added by Sanic _and_ by the ASGI handler.

Reply:
Actually, in the test this is correct because the `asgi.py` MockTransport does not differentiate between chunked or unchunked encoding, so we need this in out response.text to verify that chunked mode was used correctly by the sanic-side stream fn.

Reply:
Continuing conversation in #1964